aten/src/ATen/native/ComplexHelper.h

```diff
@@ -13,17 +13,52 @@ inline std::vector<int64_t> computeStrideForComplex(IntArrayRef oldstride) {
   return res;
 }
 
-// expects as input a complex tensor and returns back a float tensor
-// containing the complex values in the last two dimensions
-inline Tensor view_complex_as_float(const Tensor& self) {
-  TORCH_INTERNAL_ASSERT(self.is_complex());
+// expects as input a complex tensor and returns back a tensor
+// with corresponding real dtype containing the complex values
+// in the last two dimensions
+Tensor view_as_real(const Tensor& self) {
+  TORCH_CHECK(self.is_complex(), "view_as_real is only supported for complex tensors");
   auto new_sizes = self.sizes().vec();
   // last dimension will always have two elements containing the real and imag vals
   new_sizes.emplace_back(2);
-  auto new_strides = computeStrideForComplex(self.strides());
+  auto new_strides = computeStrideForViewAsReal(self.strides());
   auto new_storage_offset = 2 * self.storage_offset();
   const auto float_type = c10::toValueType(self.scalar_type());
   return at::empty({0}, self.options().dtype(float_type)).set_(self.storage(), new_storage_offset, new_sizes, new_strides);
 }
 
+inline std::vector<int64_t> computeStrideForViewAsComplex(IntArrayRef oldstride) {
```

Comment:
consider using a DimVector here

Reply:
what's the advantage of using `DimVector`?